sdk/web-pubsub/web-pubsub/src/groupClient.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  OperationOptions,
+  RestResponse,
+  RestError,
+  HttpRequestBody,
+  PipelineOptions
+} from "@azure/core-http";
+import { AzureWebPubSubServiceRestAPI as GeneratedClient } from "./generated/azureWebPubSubServiceRestAPI";
+import { createSpan } from "./tracing";
+import normalizeSendToAllOptions from "./normalizeOptions";
+import { getContentTypeForMessage } from "./utils";
+import { JSONTypes } from "./hubClient";
+
+/**
+ * Options for constructing a GroupAdmin client.
+ */
+export interface GroupAdminClientOptions extends PipelineOptions {}
+
+/**
+ * Options for adding a connection to a group.
+ */
+export interface GroupAddConnectionOptions extends OperationOptions {}
+
+/**
+ * Options for adding a user to a group.
+ */
+export interface GroupAddUserOptions extends OperationOptions {}
+
+/**
+ * Options for checking if a user is in a group
+ */
+export interface GroupHasUserOptions extends OperationOptions {}
+
+/**
+ * Options for removing a user from a group
+ */
+export interface GroupRemoveUserOptions extends OperationOptions {}
+
+/**
+ * Options for removing a connection from a group
+ */
+export interface GroupRemoveConnectionOptions extends OperationOptions {}
+
+/**
+ * Options for sending messages to a group.
+ */
+export interface GroupSendToAllOptions extends OperationOptions {
+  /**
+   * Connection ids to exclude from receiving this message.
+   */
+  excludedConnections?: string[];
+}
+
+/**
+ * Options for sending text messages to a group..
+ */
+export interface GroupSendTextToAllOptions extends OperationOptions {
+  /**
+   * Connection ids to exclude from receiving this message.
+   */
+  excludedConnections?: string[];
+  contentType: "text/plain";
+}
+
+/**
+ * Client for connecting to a Web PubSub group.
+ */
+export class WebPubSubGroup {
+  private client!: GeneratedClient;
+
+  /**
+   * The name of this group
+   */
+  public readonly groupName: string;
+
+  /**
+   * The name of the hub this group belongs to
+   */
+  public readonly hubName: string;
+
+  /**
+   * The Web PubSub API version being used by this client
+   */
+  public readonly apiVersion: string = "2020-10-01";
+
+  /**
+   * The Web PubSub endpoint this client is connected to
+   */
+  public endpoint!: string;
+
+  /**
+   * @private
+   * @internal
+   */
+  constructor(hubName: string, groupName: string) {
+    this.groupName = groupName;
+    this.hubName = hubName;
+  }
+  /**
+   * Add a specific connection to this group
+   *
+   * @param connectionId The connection id to add to this group
```

Comment:
no format requirements on `connectionId`

Reply:
Actually seems like the connection id comes from somewhere else since it can 404... where is this made?

Reply:
The service assigns the connection id which you will get when your backend is notified that a user connected.